cmd/scale.go

```diff
@@ -255,16 +288,35 @@ func (sc *scaleCmd) run(cmd *cobra.Command, args []string) error {
 		currentNodeCount = len(indexes)
 
 		if currentNodeCount == sc.newDesiredAgentCount {
-			log.Info("Cluster is currently at the desired agent count.")
+			sc.printScaleTargetEqualsExisting(currentNodeCount)
 			return nil
 		}
 		highestUsedIndex = indexes[len(indexes)-1]
 
-		// Scale down Scenario
+		// VMAS scale down scenario
 		if currentNodeCount > sc.newDesiredAgentCount {
-			if sc.masterFQDN == "" {
+			if sc.apiserverURL == "" {
 				cmd.Usage()
-				return errors.New("master-FQDN is required to scale down a kubernetes cluster's agent pool")
+				return errors.New("--apiserver (or the deprecated --master-FQDN) is required to scale down a kubernetes cluster's agent pool")
+			}
+
+			if sc.nodes != nil {
+				if len(sc.nodes) > 1 {
+					sc.logger.Infof("There are %d nodes in pool %s before scaling down to %d:\n", len(sc.nodes), sc.agentPoolToScale, sc.newDesiredAgentCount)
+				} else {
+					sc.logger.Infof("There is %d node in pool %s before scaling down to %d:\n", len(sc.nodes), sc.agentPoolToScale, sc.newDesiredAgentCount)
+				}
+				operations.PrintNodes(sc.nodes)
+				numNodesFromK8sAPI := len(sc.nodes)
+				if currentNodeCount != numNodesFromK8sAPI {
+					sc.logger.Warnf("The Azure API reports %d vms in pool %s", currentNodeCount, sc.agentPoolToScale)
+				} else {
+					if len(sc.nodes) > 1 {
```

Comment:
same thing here with the logic for 0 nodes

Reply:
although is 0 ever the case here? can `currentNodeCount-sc.newDesiredAgentCount = 0`?

Reply:
I guess not because this whole block is `currentNodeCount > sc.newDesiredAgentCount`